src/parallel/StaticIrregDomainDecomposition.cpp

```diff
@@ -0,0 +1,120 @@
+
+StaticIrregDomainDecomposition::StaticIrregDomainDecomposition(
+	Domain* domain, MPI_Comm comm, const std::array<std::vector<unsigned int>, DIMgeom>& subdomainWeights) : 
+		DomainDecomposition(comm, {(int)subdomainWeights[0].size(), (int)subdomainWeights[1].size(), (int)subdomainWeights[2].size()}), 
+		_subdomainWeights(subdomainWeights),
+		_boxMin{0},
+		_boxMax{0},
+		_domainLength{domain->getGlobalLength(0), domain->getGlobalLength(1), domain->getGlobalLength(2)} {
+			if(_subdomainWeights[0].size() == 0 && _subdomainWeights[1].size() == 0 && _subdomainWeights[2].size() == 0) { //default behaviour, regular grid
+				for (int i = 0; i < 3; i++) {
+					for (int j = 0; j < _gridSize[i]; j++) {
+						_subdomainWeights[i].push_back(1);  //equal subdomains, so weight = 1
+					}
+				}
+			}
+			updateSubdomainDimensions();
+		}
+
+void StaticIrregDomainDecomposition::readXML(XMLfileUnits& xmlconfig) {
+	DomainDecompMPIBase::readXML(xmlconfig);
+	//bypass DomainDecomposition readXML to avoid reading MPIGridDims
+
+	if(xmlconfig.changecurrentnode("subdomainSizeCSV")) {
+		std::string filename = xmlconfig.getNodeValue_string("filename");
+		updateSubdomainWeightsFromFile(filename);
+		_gridSize[0] = (int)_subdomainWeights[0].size(); //_gridSize still contains the number of ranks per dimension, just not the actual "size" of subdomains
+		_gridSize[1] = (int)_subdomainWeights[1].size();
+		_gridSize[2] = (int)_subdomainWeights[2].size();
+		xmlconfig.changecurrentnode("..");
+		initMPIGridDims();							//to recalculate _coords
+		updateSubdomainDimensions();				//recalculate sizes from _coords
+	}
+}
+
+double StaticIrregDomainDecomposition::getBoundingBoxMin(int dimension, Domain* domain) {
+	return _boxMin[dimension];
+}
+
+double StaticIrregDomainDecomposition::getBoundingBoxMax(int dimension, Domain* domain) {
+	return _boxMax[dimension];
+}
+
+void StaticIrregDomainDecomposition::updateSubdomainDimensions() {
+	std::array<int, 3> totWeight{0}; //stores the total weight across the dimensions
+	for(int i = 0; i < 3; i++) {
+		for(unsigned int weight: _subdomainWeights[i]) {
+			totWeight[i] += weight;
+		}
+	}
+	Log::global_log->debug() << "totweight: " << totWeight[0] << " " << totWeight[1] << " " << totWeight[2]  << std::endl;
+
+	int backWeight; //stores the cumulative weights of all subdomains previous to current subdomain
+	for(int i = 0; i < 3; i++) {
+		backWeight = 0;
+		for(int j = 0; j < _coords[i]; j++) {
+			backWeight += _subdomainWeights[i][j];
+		}
+		Log::global_log->debug() << "backweight: " << i << " " << backWeight << " coords: " << _coords[0] << ", " <<_coords[1] << ", " << _coords[2] << std::endl;
+
+		//calculate box bounds from cumulative weights of previous ranks, and the weight of the current rank
+		_boxMin[i] = static_cast<double>(backWeight) * _domainLength[i] / totWeight[i];
+		_boxMax[i] = _boxMin[i] + (static_cast<double>(_subdomainWeights[i][_coords[i]]) * _domainLength[i] / totWeight[i]);
+	}
+}
```

Comment:
```suggestion
	for(int i = 0; i < 3; i++) {
		const auto backWeight = std::reduce(_subdomainWeights[i].begin(), _subdomainWeights[i].begin() + _coords[i], 0u);
		const auto totalWeight = std::reduce(_subdomainWeights[i].begin() + _coords[i], _subdomainWeights[i].end(), backWeight);
		Log::global_log->debug() << "Dim: " << i << " totalWeight: " << totalWeight << " backWeight: " << backWeight << " coords: " << _coords[0] << ", " <<_coords[1] << ", " << _coords[2]std::endl;

		//calculate box bounds from cumulative weights of previous ranks, and the weight of the current rank
		_boxMin[i] = static_cast<double>(backWeight) * _domainLength[i] / totalWeight;
		_boxMax[i] = _boxMin[i] + (static_cast<double>(_subdomainWeights[i][_coords[i]]) * _domainLength[i] / totalWeight);
	}
}
```

- [ ] Be more verbose with variable names. Avoid abbreviations as much as possible.
- [ ] `std::accumulate` or `std::reduce` allows for `const` initialization and makes for easier reading because there are fewer degrees of freedom.
- [ ] `totWeight` doesn't need to be an array, right?